Integrations/FeedTAXII/FeedTAXII.py

```diff
@@ -0,0 +1,918 @@
+            if htype not in ['md5', 'sha1', 'sha256', 'ssdeep']:
+                continue
+
+            value = h.find('Simple_Hash_Value')
+            if value is None:
+                continue
+            value = value.string.lower()
+
+            result.append({
+                'indicator': value,
+                'htype': htype,
+                'type': 'File'
+            })
+
+        for r in result:
+            for r2 in result:
+                if r['htype'] == r2['htype']:
+                    continue
+
+                r['stix_file_{}'.format(r2['htype'])] = r2['indicator']
+
+            r.update(bprops)
+
+        return result
+
+
+''' uriobject.py '''
+
+
+class URIObject(object):
+    @staticmethod
+    def decode(props, **kwargs):
+        utype = props.get('type', 'URL')
+        if utype == 'URL':
+            type_ = 'URL'
+        elif utype == 'Domain Name':
+            type_ = 'Domain'
+        else:
+            return []
+
+        url = props.find('Value')
+        if url is None:
+            return []
+
+        return [{
+            'indicator': url.string.encode('utf8', 'replace').decode(),
+            'type': type_
+        }]
+
+
+class SocketAddressObject(object):
+    @staticmethod
+    def decode(props, **kwargs):
+        ip = props.get('ip_address', None)
+        if ip:
+            return AddressObject.decode(ip)
+        return []
+
+
+class LinkObject(object):
+    @staticmethod
+    def decode(props, **kwargs):
+        ltype = props.get('type', 'URL')
+        if ltype != 'URL':
+            LOG('Unhandled LinkObjectType type: {}'.format(ltype))
+            return []
+        value = props.get('value', None)
+        if value is None:
+            LOG('no value in observable LinkObject')
+            return []
+        if not isinstance(value, string_types):
+            value = value.get('value', None)
+            if value is None:
+                LOG('no value in observable LinkObject')
+                return []
+        return [{
+            'indicator': value,
+            'type': ltype
+        }]
+
+
+class HTTPSessionObject(object):
+    @staticmethod
+    def decode(props, **kwargs):
+        if 'http_request_response' in props.keys():
+            tmp = props['http_request_response']
+
+            if len(tmp) == 1:
+                item = tmp[0]
+                http_client_request = item.get('http_client_request', None)
+                if http_client_request is not None:
+                    http_request_header = http_client_request.get('http_request_header', None)
+                    if http_request_header is not None:
+                        raw_header = http_request_header.get('raw_header', None)
+                        if raw_header is not None:
+                            return [{
+                                'indicator': raw_header.split('\n')[0],
+                                'type': 'http-session',  # we don't support this type natively in demisto
```

Comment:
what will happen if we push an indicator that dose not exist?

Reply:
`demisto.createIndicators` will drop it (i.e. not create the indicator)

Reply:
The server will drop it.
Since it wasn't yet decided if we'll support this type or not, I kept it